src/mini/ini.h

```diff
@@ -96,6 +96,18 @@
 
 namespace mINI
 {
+	struct mINIFilePath 
+	{
+		std::filesystem::path value;
+		explicit mINIFilePath(std::filesystem::path const& p) : value(p) {}
+
+		static mINIFilePath fromString(std::string const& str)
```

Comment:
Was wondering if the fromString static method could be an overloaded constructor taking a string path instead? It may make the calling convention less verbose for example. For instance `mINIFilePath(filename)` instead of `mINIFilePath::fromString(filename)`

Reply:
Yeah I'm gonna need to simplify this. I don't think there's a need for a wrapper.

Reply:
> Yeah I'm gonna need to simplify this. I don't think there's a need for a wrapper.

Wrapper is needed if you want it to be 100% backwards compatible with all existing code. For example, if someone added conversion operator to std::string to some class: `struct test
{
    int val;

    operator std::string() const
    {
        return std::to_string(val);
    }
};`  Then they won't be able to pass it to the function accepting std::filesystem::path. And you can't have function accepting both, std::string and std::filesystem::path directly either, because you will get compiler error when trying to pass a string directly, since string can be implicitly convert to filesysyem::path, and it won't know which overload to use. Try it yourself.

Reply:
> Was wondering if the fromString static method could be an overloaded constructor taking a string path instead? It may make the calling convention less verbose for example. For instance `mINIFilePath(filename)` instead of `mINIFilePath::fromString(filename)`

You can't have 1 constructor accepting std::string and the other std::Filesystem::path, since string by itself is implicitly convertible to std::filesystem::path, so it won't know which overload to use, and it will result in a compile error.

Reply:
Tnx for insights, I'll experiment with it when I have some time on my hands.